pkg/mimirpb/compat.go

```diff
@@ -47,19 +47,12 @@ func NewWriteRequest(metadata []*MetricMetadata, source WriteRequest_SourceEnum)
 // AddFloatSeries converts matched slices of Labels, Samples, Exemplars into a WriteRequest
 // proto. It gets timeseries from the pool, so ReuseSlice() should be called when done. Note that this
 // method implies that only a single sample and optionally exemplar can be set for each series.
-func (req *WriteRequest) AddFloatSeries(lbls []labels.Labels, samples []Sample, exemplars []*Exemplar) *WriteRequest {
+func (req *WriteRequest) AddFloatSeries(lbls []labels.Labels, samples []Sample, exemplars []Exemplar) *WriteRequest {
 	for i, s := range samples {
 		ts := TimeseriesFromPool()
 		ts.Labels = append(ts.Labels, FromLabelsToLabelAdapters(lbls[i])...)
 		ts.Samples = append(ts.Samples, s)
-
-		if exemplars != nil {
-			// If provided, we expect a matched entry for exemplars (like labels and samples) but the
-			// entry may be nil since not every timeseries is guaranteed to have an exemplar.
-			if e := exemplars[i]; e != nil {
-				ts.Exemplars = append(ts.Exemplars, *e)
-			}
-		}
+		ts.Exemplars = append(ts.Exemplars, exemplars...)
```

Comment:
This adds same exemplars to each series. That's now how this function is supposed to work.

Idea of this function is that each series gets single sample and exemplar (if not nil).

Reply:
I see now, thanks. I'll try to find a workaround (:star: Optional? :star:) and also will cover this with a test.